diff -Naurw ./arch/arm/lib/relocate.S ../git/arch/arm/lib/relocate.S
--- ./arch/arm/lib/relocate.S	2019-07-16 16:01:44.474024579 +0200
+++ ../git/arch/arm/lib/relocate.S	2019-07-16 16:02:13.844444159 +0200
@@ -78,6 +78,8 @@
  */
 
 ENTRY(relocate_code)
+	/* b   relocate_code */
+
 	ldr	r1, =__image_copy_start	/* r1 <- SRC &__image_copy_start */
 	subs	r4, r0, r1		/* r4 <- relocation offset */
 	beq	relocate_done		/* skip relocation */
diff -Naurw ./drivers/mmc/sdhci.c ../git/drivers/mmc/sdhci.c
--- ./drivers/mmc/sdhci.c	2019-07-16 16:01:44.750009700 +0200
+++ ../git/drivers/mmc/sdhci.c	2019-07-16 16:02:13.844444159 +0200
@@ -254,6 +254,8 @@
 		sdhci_writeb(host, 0xe, SDHCI_TIMEOUT_CONTROL);
 	}
 
+	sdhci_writeb(host, sdhci_readb(host, 0x204) | 0x80, 0x204 );
+
 	sdhci_writel(host, cmd->cmdarg, SDHCI_ARGUMENT);
 #ifdef CONFIG_MMC_SDHCI_SDMA
 	if (data) {
diff -Naurw ./include/configs/sama5d27_som1_ek.h ../git/include/configs/sama5d27_som1_ek.h
--- ./include/configs/sama5d27_som1_ek.h	2019-07-16 16:01:44.862003663 +0200
+++ ../git/include/configs/sama5d27_som1_ek.h	2019-07-16 16:02:13.844444159 +0200
@@ -12,7 +12,7 @@
 #include "at91-sama5_common.h"
 
 #undef CONFIG_SYS_AT91_MAIN_CLOCK
-#define CONFIG_SYS_AT91_MAIN_CLOCK      24000000 /* from 24 MHz crystal */
+#define CONFIG_SYS_AT91_MAIN_CLOCK      12000000 /* from 12 MHz crystal */
 
 /* SDRAM */
 #define CONFIG_SYS_SDRAM_BASE		0x20000000
diff -Naurw ./Makefile ../git/Makefile
--- ./Makefile	2019-07-16 16:01:44.382029537 +0200
+++ ../git/Makefile	2019-07-16 16:02:23.855906754 +0200
@@ -254,7 +254,7 @@
 
 HOSTCC       = cc
 HOSTCXX      = c++
-HOSTCFLAGS   = -Wall -Wstrict-prototypes -O2 -fomit-frame-pointer \
+HOSTCFLAGS   = -Wall -Wstrict-prototypes -O1 -g -fomit-frame-pointer \
 		$(if $(CONFIG_TOOLS_DEBUG),-g)
 HOSTCXXFLAGS = -O2
 
