diff -Naurw ./arch/arm/lib/relocate.S ../git.new/arch/arm/lib/relocate.S
--- ./arch/arm/lib/relocate.S	2018-11-13 10:44:21.000000000 +0100
+++ ../git.new/arch/arm/lib/relocate.S	2019-02-27 13:54:54.711544616 +0100
@@ -77,6 +77,8 @@
  */
 
 ENTRY(relocate_code)
+	/* b   relocate_code */
+
 	ldr	r1, =__image_copy_start	/* r1 <- SRC &__image_copy_start */
 	subs	r4, r0, r1		/* r4 <- relocation offset */
 	beq	relocate_done		/* skip relocation */
diff -Naurw ./drivers/mmc/sdhci.c ../git.new/drivers/mmc/sdhci.c
--- ./drivers/mmc/sdhci.c	2018-11-13 10:44:22.000000000 +0100
+++ ../git.new/drivers/mmc/sdhci.c	2019-02-27 13:54:54.555544616 +0100
@@ -254,6 +254,8 @@
 		sdhci_writeb(host, 0xe, SDHCI_TIMEOUT_CONTROL);
 	}
 
+	sdhci_writeb(host, sdhci_readb(host, 0x204) | 0x80, 0x204 );
+
 	sdhci_writel(host, cmd->cmdarg, SDHCI_ARGUMENT);
 #ifdef CONFIG_MMC_SDHCI_SDMA
 	if (data) {
diff -Naurw ./include/configs/sama5d27_som1_ek.h ../git.new/include/configs/sama5d27_som1_ek.h
--- ./include/configs/sama5d27_som1_ek.h	2018-11-13 10:44:22.000000000 +0100
+++ ../git.new/include/configs/sama5d27_som1_ek.h	2019-02-27 13:54:54.611544616 +0100
@@ -12,7 +12,7 @@
 #include "at91-sama5_common.h"
 
 #undef CONFIG_SYS_AT91_MAIN_CLOCK
-#define CONFIG_SYS_AT91_MAIN_CLOCK      24000000 /* from 24 MHz crystal */
+#define CONFIG_SYS_AT91_MAIN_CLOCK      12000000 /* from 12 MHz crystal */
 
 #define CONFIG_MISC_INIT_R
 
diff -Naurw ./Makefile ../git.new/Makefile
--- ./Makefile	2018-11-13 10:44:21.000000000 +0100
+++ ../git.new/Makefile	2019-02-27 13:54:54.519544616 +0100
@@ -254,7 +254,8 @@
 
 HOSTCC       = cc
 HOSTCXX      = c++
-HOSTCFLAGS   = -Wall -Wstrict-prototypes -O2 -fomit-frame-pointer \
+/* HOSTCFLAGS   = -Wall -Wstrict-prototypes -O2 -fomit-frame-pointer \ */
+HOSTCFLAGS   = -Wall -Wstrict-prototypes -O1 -g -fomit-frame-pointer \
 		$(if $(CONFIG_TOOLS_DEBUG),-g)
 HOSTCXXFLAGS = -O2
 
