diff -Naurw ./board/sama5d27_som1_ek/sama5d27_som1_ek.h ../git/board/sama5d27_som1_ek/sama5d27_som1_ek.h
--- ./board/sama5d27_som1_ek/sama5d27_som1_ek.h	2019-07-16 15:52:56.843876318 +0200
+++ ../git/board/sama5d27_som1_ek/sama5d27_som1_ek.h	2019-07-16 15:54:23.618710766 +0200
@@ -31,10 +31,10 @@
 /*
  * PMC Settings
  */
-#define BOARD_MAINOSC		24000000
+#define BOARD_MAINOSC		12000000
 
-/* PCK: 492MHz, MCK: 164MHz */
-#define BOARD_PLLA_MULA		40
+/* PCK: 498MHz, MCK: 166MHz */
+#define BOARD_PLLA_MULA		82
 
 #define BOARD_PCK		((unsigned long)((BOARD_MAINOSC * (BOARD_PLLA_MULA + 1)) / 2))
 #define BOARD_MCK		((unsigned long)((BOARD_MAINOSC * (BOARD_PLLA_MULA + 1)) / 2 / 3))
@@ -53,7 +53,7 @@
 					| AT91C_PMC_MDIV_3 \
 					| AT91C_PMC_CSS_PLLA_CLK)
 
-#define MASTER_CLOCK		164000000
+#define MASTER_CLOCK		166000000
 
 #define PLLA_SETTINGS		(BOARD_CKGR_PLLA | \
 				BOARD_PLLACOUNT | \
diff -Naurw ./board/sama5d27_som1_ek/sama5d27_som1_eksd_uboot_defconfig ../git/board/sama5d27_som1_ek/sama5d27_som1_eksd_uboot_defconfig
--- ./board/sama5d27_som1_ek/sama5d27_som1_eksd_uboot_defconfig	2019-07-16 15:52:56.843876318 +0200
+++ ../git/board/sama5d27_som1_ek/sama5d27_som1_eksd_uboot_defconfig	2019-07-16 15:54:23.618710766 +0200
@@ -1,6 +1,7 @@
 CONFIG_SAMA5D27_SOM1_EK=y
 CONFIG_RAM_128MB=y
 CONFIG_SDCARD=y
+CONFIG_SDHC1=y
 CONFIG_JUMP_ADDR="0x23f00000"
 CONFIG_DEBUG=y
 # CONFIG_ENTER_NWD is not set
diff -Naurw ./driver/sdhc.c ../git/driver/sdhc.c
--- ./driver/sdhc.c	2019-07-16 15:52:56.855875594 +0200
+++ ../git/driver/sdhc.c	2019-07-16 15:54:23.622710531 +0200
@@ -496,6 +496,8 @@
 	return 0;
 }
 
+#if 0
+
 static int sdhc_is_card_inserted(struct sd_card *sdcard)
 {
 	/*
@@ -542,6 +544,13 @@
 	return is_inserted;
 }
 
+#else
+
+#define sdhc_is_card_inserted( x )  1
+
+#endif
+
+
 static int sdhc_init(struct sd_card *sdcard)
 {
 	unsigned int normal_status_mask, error_status_mask;
@@ -750,6 +759,8 @@
 		}
 	}
 
+	sdhc_writeb( SDMMC_MC1R, sdhc_readb(SDMMC_MC1R) | 0x80 );
+
 	sdhc_writel(SDMMC_ARG1R, sd_cmd->argu);
 
 	sdhc_writew(SDMMC_CR, cmd_reg);
