diff -Naurw ./mpio/gpio.py ../git/mpio/gpio.py
--- ./mpio/gpio.py	2019-11-07 16:12:32.552061604 +0100
+++ ../git/mpio/gpio.py	2019-11-07 16:20:33.785750494 +0100
@@ -255,7 +255,7 @@
         req.lineoffsets[0] = self._line_offset
         req.default_values[0] = 0
         req.flags = _GPIOHANDLE_REQUEST_INPUT
-        req.consumer_label = "MPIO"
+        req.consumer_label = b"MPIO"
         req.lines = 1
         req.fd = 0
 
@@ -287,7 +287,7 @@
         req.lineoffsets[0] = self._line_offset
         req.default_values[0] = 0
         req.flags = _GPIOHANDLE_REQUEST_OUTPUT
-        req.consumer_label = "MPIO"
+        req.consumer_label = b"MPIO"
         req.lines = 1
         req.fd = 0
 
@@ -329,7 +329,7 @@
         req = _CGPIOGPIOEventRequest()
         req.lineoffset = self._line_offset
         req.handleflags = 0
-        req.consumer_label = "MPIO"
+        req.consumer_label = b"MPIO"
         if edge == self.BOTH:
             req.eventflags = _GPIOEVENT_REQUEST_RISING_EDGE | \
                              _GPIOEVENT_REQUEST_FALLING_EDGE
