diff -Naurw ./setup.py ../cryptoauthlib-20190903/setup.py
--- ./setup.py	2019-10-09 14:16:49.430744848 +0200
+++ ../cryptoauthlib-20190903/setup.py	2019-10-09 14:17:12.417589633 +0200
@@ -205,7 +205,7 @@
         include_package_data=True,
         distclass=BinaryDistribution,
         cmdclass=_COMMANDS,
-        setup_requires=['setuptools>=38.6.0', 'wheel'],
+        setup_requires=['setuptools>=38.6.0'],
         install_requires=['enum34;python_version<"3.4"'],
         ext_modules=_EXTENSIONS,
         python_requires='>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, <4',
