diff -Naurw ./arch/arm/boot/dts/at91-sama5d27_som1.dtsi ../kernel-source/arch/arm/boot/dts/at91-sama5d27_som1.dtsi
--- ./arch/arm/boot/dts/at91-sama5d27_som1.dtsi	2019-07-16 17:58:54.720991876 +0200
+++ ../kernel-source/arch/arm/boot/dts/at91-sama5d27_som1.dtsi	2019-07-17 10:52:40.958276631 +0200
@@ -19,7 +19,7 @@
 		};
 
 		main_xtal {
-			clock-frequency = <24000000>;
+			clock-frequency = <12000000>;
 		};
 	};
 
diff -Naurw ./arch/arm/boot/dts/at91-sama5d27_som1_ek.dts ../kernel-source/arch/arm/boot/dts/at91-sama5d27_som1_ek.dts
--- ./arch/arm/boot/dts/at91-sama5d27_som1_ek.dts	2019-07-16 17:58:54.720991876 +0200
+++ ../kernel-source/arch/arm/boot/dts/at91-sama5d27_som1_ek.dts	2019-07-17 17:54:30.798099282 +0200
@@ -23,6 +23,7 @@
 		i2c0	= &i2c0;
 		i2c1	= &i2c1;
 		i2c2	= &i2c2;
+		i2c3	= &i2c3;
 	};
 
 	chosen {
@@ -53,16 +54,27 @@
 		};
 
 		sdmmc0: sdio-host@a0000000 {
-			bus-width = <8>;
 			pinctrl-names = "default";
 			pinctrl-0 = <&pinctrl_sdmmc0_default>;
+			non-removable;
 			status = "okay";
+			wilc_sdio@0{
+				compatible = "microchip,wilc1000", "microchip,wilc3000";
+				/* WILC1000 doesn't handle reset-gpio properly    */
+				/* reset-gpios = <&pioA PIN_PA7 GPIO_ACTIVE_LOW>; */
+				/* chip_en-gpios = <&pioA PIN_PA10 GPIO_ACTIVE_HIGH>; */
+				irq-gpios = <&pioA PIN_PA11 GPIO_ACTIVE_HIGH>;
+				status = "okay";
+				reg = <0>;
+				bus-width = <4>;
+			};
 		};
 
 		sdmmc1: sdio-host@b0000000 {
 			bus-width = <4>;
 			pinctrl-names = "default";
 			pinctrl-0 = <&pinctrl_sdmmc1_default>;
+			cd-gpios = <&pioA PIN_PD23 GPIO_ACTIVE_HIGH>;
 			status = "okay";
 		};
 
@@ -89,7 +101,7 @@
 			isc: isc@f0008000 {
 				pinctrl-names = "default";
 				pinctrl-0 = <&pinctrl_isc_base &pinctrl_isc_data_8bit &pinctrl_isc_data_9_10 &pinctrl_isc_data_11_12>;
-				status = "okay";
+				status = "disabled"; /* conflict with leds */
 			};
 
 			qspi1: spi@f0024000 {
@@ -131,7 +143,7 @@
 				pinctrl-0 = <&pinctrl_mikrobus2_uart>;
 				atmel,use-dma-rx;
 				atmel,use-dma-tx;
-				status = "okay";
+				status = "disabled"; /* conflict with sdmmc1_cd */
 			};
 
 			i2c0: i2c@f8028000 {
@@ -147,6 +159,48 @@
 				status = "disabled"; /* Conflict with leds. */
 			};
 
+			flx0: flexcom@f8034000 {
+				atmel,flexcom-mode = <ATMEL_FLEXCOM_MODE_TWI>;
+				status = "okay";
+
+				i2c2: i2c@600 {
+					compatible = "atmel,sama5d2-i2c";
+					reg = <0x600 0x200>;
+					interrupts = <19 IRQ_TYPE_LEVEL_HIGH 7>;
+					dmas = <0>, <0>;
+					dma-names = "tx", "rx";
+					#address-cells = <1>;
+					#size-cells = <0>;
+					clocks = <&flx0_clk>;
+					pinctrl-names = "default";
+					pinctrl-0 = <&pinctrl_flx0_default>;
+					atmel,fifo-size = <16>;
+					status = "okay";
+				};
+			};
+
+			flx3: flexcom@fc014000 {
+				atmel,flexcom-mode = <ATMEL_FLEXCOM_MODE_TWI>;
+				status = "okay";
+
+				i2c3: i23@600 {
+					compatible = "atmel,sama5d2-i2c";
+					reg = <0x600 0x200>;
+					interrupts = <22 IRQ_TYPE_LEVEL_HIGH 7>;
+					dmas = <0>, <0>;
+					dma-names = "tx", "rx";
+					#address-cells = <1>;
+					#size-cells = <0>;
+					clocks = <&flx3_clk>;
+					clock-names = "spi_clk";
+					pinctrl-names = "default";
+					pinctrl-0 = <&pinctrl_flx3_default>;
+					atmel,fifo-size = <16>;
+					status = "okay";
+				};
+			};
+
+#if 0
 			flx1: flexcom@f8038000 {
 				atmel,flexcom-mode = <ATMEL_FLEXCOM_MODE_TWI>;
 				status = "okay";
@@ -163,10 +217,10 @@
 					pinctrl-names = "default";
 					pinctrl-0 = <&pinctrl_mikrobus_i2c>;
 					atmel,fifo-size = <16>;
-					status = "okay";
+					status = "disabled";
 				};
 			};
-
+#endif
 			shdwc@f8048010 {
 				atmel,shdwc-debouncer = <976>;
 				atmel,wakeup-rtc-timer;
@@ -194,9 +248,10 @@
 				atmel,use-dma-tx;
 				pinctrl-names = "default";
 				pinctrl-0 = <&pinctrl_mikrobus1_uart>;
-				status = "okay";
+				status = "disabled";
 			};
 
+#if 0
 			flx3: flexcom@fc014000 {
 				atmel,flexcom-mode = <ATMEL_FLEXCOM_MODE_SPI>;
 				status = "disabled";
@@ -251,7 +306,7 @@
 					pinctrl-names = "default";
 					pinctrl-0 = <&pinctrl_mikrobus_spi &pinctrl_mikrobus1_spi_cs &pinctrl_mikrobus2_spi_cs>;
 					atmel,fifo-size = <16>;
-					status = "okay"; /* Conflict with uart6 and i2c3. */
+					status = "disabled"; /* Conflict with uart6 and i2c3. */
 				};
 
 				i2c3: i2c@600 {
@@ -269,6 +324,7 @@
 					status = "disabled"; /* Conflict with uart6 and spi3. */
 				};
 			};
+#endif
 
 			i2c1: i2c@fc028000 {
 				dmas = <0>, <0>;
@@ -292,12 +348,15 @@
 					bias-disable;
 				};
 
+				pinctrl_flx0_default: flx0_default {
+					pinmux = <PIN_PB28__FLEXCOM0_IO0>,
+						 <PIN_PB29__FLEXCOM0_IO1>;
+					bias-disable;
+				};
+
 				pinctrl_flx3_default: flx3_default {
-					pinmux = <PIN_PC20__FLEXCOM3_IO0>,
-						 <PIN_PC19__FLEXCOM3_IO1>,
-						 <PIN_PC18__FLEXCOM3_IO2>,
-						 <PIN_PC21__FLEXCOM3_IO3>,
-						 <PIN_PC22__FLEXCOM3_IO4>;
+					pinmux = <PIN_PB23__FLEXCOM3_IO0>,
+						 <PIN_PB22__FLEXCOM3_IO1>;
 					bias-disable;
 				};
 
@@ -351,9 +410,10 @@
 				};
 
 				pinctrl_led_gpio_default: led_gpio_default {
-					pinmux = <PIN_PA10__GPIO>,
-						 <PIN_PB1__GPIO>,
-						 <PIN_PA31__GPIO>;
+					pinmux = <PIN_PC12__GPIO>,
+							<PIN_PC13__GPIO>,
+							<PIN_PC15__GPIO>,
+							<PIN_PC17__GPIO>;
 					bias-pull-up;
 				};
 
@@ -363,20 +423,22 @@
 							 <PIN_PA2__SDMMC0_DAT0>,
 							 <PIN_PA3__SDMMC0_DAT1>,
 							 <PIN_PA4__SDMMC0_DAT2>,
-							 <PIN_PA5__SDMMC0_DAT3>,
-							 <PIN_PA6__SDMMC0_DAT4>,
-							 <PIN_PA7__SDMMC0_DAT5>,
-							 <PIN_PA8__SDMMC0_DAT6>,
-							 <PIN_PA9__SDMMC0_DAT7>;
+							 <PIN_PA5__SDMMC0_DAT3>;
+						bias-disable;
+					};
+
+					conf-ck {
+						pinmux = <PIN_PA0__SDMMC0_CK>;
 						bias-disable;
 					};
 
-					ck_cd_vddsel {
-						pinmux = <PIN_PA0__SDMMC0_CK>,
-							 <PIN_PA11__SDMMC0_VDDSEL>,
-							 <PIN_PA13__SDMMC0_CD>;
+					conf-wilc {
+						pinmux = <PIN_PA10__GPIO>,
+							<PIN_PA7__GPIO>,
+							<PIN_PA11__GPIO>;
 						bias-disable;
 					};
+
 				};
 
 				pinctrl_sdmmc1_default: sdmmc1_default {
@@ -391,7 +453,7 @@
 
 					conf-ck_cd {
 						pinmux = <PIN_PA22__SDMMC1_CK>,
-							 <PIN_PA30__SDMMC1_CD>;
+							 <PIN_PD23__GPIO>;
 						bias-disable;
 					};
 				};
@@ -538,21 +600,26 @@
 		compatible = "gpio-leds";
 		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_led_gpio_default>;
-		status = "okay"; /* Conflict with pwm0. */
+		status = "okay";
+
+		user {
+			label = "user";
+			gpios = <&pioA PIN_PC12 GPIO_ACTIVE_HIGH>;
+		};
 
-		red {
-			label = "red";
-			gpios = <&pioA PIN_PA10 GPIO_ACTIVE_HIGH>;
+		wifi {
+			label = "wifi";
+			gpios = <&pioA PIN_PC13 GPIO_ACTIVE_HIGH>;
 		};
 
-		green {
-			label = "green";
-			gpios = <&pioA PIN_PB1 GPIO_ACTIVE_HIGH>;
+		bluetooth {
+			label = "bluetooth";
+			gpios = <&pioA PIN_PC15 GPIO_ACTIVE_HIGH>;
 		};
 
-		blue {
-			label = "blue";
-			gpios = <&pioA PIN_PA31 GPIO_ACTIVE_HIGH>;
+		zigbee {
+			label = "zigbee";
+			gpios = <&pioA PIN_PC17 GPIO_ACTIVE_HIGH>;
 			linux,default-trigger = "heartbeat";
 		};
 	};
diff -Naurw ./drivers/mmc/core/sdio.c ../kernel-source/drivers/mmc/core/sdio.c
--- ./drivers/mmc/core/sdio.c	2019-07-16 17:58:55.316956489 +0200
+++ ../kernel-source/drivers/mmc/core/sdio.c	2019-07-17 10:52:40.958276631 +0200
@@ -1102,6 +1102,8 @@
 	u32 ocr, rocr;
 	struct mmc_card *card;
 
+    printk( KERN_ALERT "**** mmc_attach_sdio( %s )\n", mmc_hostname( host ) );
+
 	WARN_ON(!host->claimed);
 
 	err = mmc_send_io_op_cond(host, 0, &ocr);
diff -Naurw ./drivers/mmc/host/sdhci-of-at91.c ../kernel-source/drivers/mmc/host/sdhci-of-at91.c
--- ./drivers/mmc/host/sdhci-of-at91.c	2019-07-16 17:58:55.316956489 +0200
+++ ../kernel-source/drivers/mmc/host/sdhci-of-at91.c	2019-07-17 10:52:40.958276631 +0200
@@ -126,7 +126,7 @@
 {
 	sdhci_reset(host, mask);
 
-	if (host->mmc->caps & MMC_CAP_NONREMOVABLE)
+	/* if (host->mmc->caps & MMC_CAP_NONREMOVABLE) */
 		sdhci_at91_set_force_card_detect(host);
 }
 
@@ -413,7 +413,7 @@
 	 * This bit is reset when a software reset for all command is performed
 	 * so we need to implement our own reset function to set back this bit.
 	 */
-	if (host->mmc->caps & MMC_CAP_NONREMOVABLE)
+	/* if (host->mmc->caps & MMC_CAP_NONREMOVABLE) */
 		sdhci_at91_set_force_card_detect(host);
 
 	device_init_wakeup(&pdev->dev, true);
