diff -Naurw ./volatiles ../1.0-r155/volatiles
--- ./volatiles	2020-01-08 13:00:19.899797518 +0100
+++ ../1.0-r155/volatiles	2020-01-08 13:00:45.158442178 +0100
@@ -29,7 +29,6 @@
 l root root 1777 /var/lock /run/lock
 l root root 0755 /var/run /run
 l root root 1777 /var/tmp /var/volatile/tmp
-l root root 1777 /tmp /var/tmp
 d root root 0755 /var/lock/subsys none
 f root root 0664 /var/log/wtmp none
 f root root 0664 /var/run/utmp none
