diff -Naurw ./conf/machine/sama5d27-hd96.conf ../meta-atmel/conf/machine/sama5d27-hd96.conf
--- ./conf/machine/sama5d27-hd96.conf	1970-01-01 01:00:00.000000000 +0100
+++ ../meta-atmel/conf/machine/sama5d27-hd96.conf	2019-09-17 14:50:58.387363066 +0200
@@ -0,0 +1,27 @@
+#@TYPE: Machine
+#@Name: ARROW HD96 SD
+#@DESCRIPTION: Machine configuration for Arrow's HD96 board
+
+require include/sama5d2.inc
+
+MACHINE_FEATURES = "apm alsa ext2 ext3 usbhost usbgadget touchscreen"
+KERNEL_DEVICETREE = " \
+		sama5d27_hd96.dtb \
+		"
+
+do_image_wic[depends] += "u-boot-at91:do_deploy"
+IMAGE_FSTYPES += " tar.gz wic"
+WKS_FILE = "sdimage-bootpart.wks"
+
+IMAGE_BOOT_FILES = "BOOT.BIN u-boot.bin uboot.env sama5d27_hd96.itb \
+                    "
+
+UBOOT_MACHINE ?= "sama5d27_hd96_mmc1_defconfig"
+UBOOT_ENTRYPOINT = "0x20008000"
+UBOOT_LOADADDRESS = "0x20008000"
+UBOOT_ENV_SIZE = "0x4000"
+
+AT91BOOTSTRAP_MACHINE ?= "sama5d27_hd96"
+
+# Needed for FIT
+MACHINE_ESSENTIAL_EXTRA_RDEPENDS = " dt-overlay-at91"
diff -Naurw ./recipes-bsp/at91bootstrap/at91bootstrap_3.8.13.bb ../meta-atmel/recipes-bsp/at91bootstrap/at91bootstrap_3.8.13.bb
--- ./recipes-bsp/at91bootstrap/at91bootstrap_3.8.13.bb	2019-09-17 14:48:46.011177845 +0200
+++ ../meta-atmel/recipes-bsp/at91bootstrap/at91bootstrap_3.8.13.bb	2019-09-17 14:50:58.387363066 +0200
@@ -2,7 +2,7 @@
 
 LIC_FILES_CHKSUM = "file://main.c;endline=27;md5=a2a70db58191379e2550cbed95449fbd"
 
-COMPATIBLE_MACHINE = '(sama5d3xek|sama5d3-xplained|sama5d3-xplained-sd|at91sam9x5ek|at91sam9rlek|at91sam9m10g45ek|sama5d4ek|sama5d4-xplained|sama5d4-xplained-sd|sama5d2-xplained|sama5d2-xplained-sd|sama5d2-xplained-emmc|sama5d2-ptc-ek|sama5d2-ptc-ek-sd|sama5d27-som1-ek|sama5d27-som1-ek-sd|sama5d2-icp-sd)'
+COMPATIBLE_MACHINE = '(sama5d3xek|sama5d3-xplained|sama5d3-xplained-sd|at91sam9x5ek|at91sam9rlek|at91sam9m10g45ek|sama5d4ek|sama5d4-xplained|sama5d4-xplained-sd|sama5d2-xplained|sama5d2-xplained-sd|sama5d2-xplained-emmc|sama5d2-ptc-ek|sama5d2-ptc-ek-sd|sama5d27-som1-ek|sama5d27-som1-ek-sd|sama5d2-icp-sd|sama5d27-hd96)'
 
 SRC_URI = "git://github.com/linux4sam/at91bootstrap.git;protocol=https \
 "
diff -Naurw ./recipes-bsp/at91bootstrap/at91bootstrap_git.bb ../meta-atmel/recipes-bsp/at91bootstrap/at91bootstrap_git.bb
--- ./recipes-bsp/at91bootstrap/at91bootstrap_git.bb	2019-09-17 14:48:46.011177845 +0200
+++ ../meta-atmel/recipes-bsp/at91bootstrap/at91bootstrap_git.bb	2019-09-17 14:50:58.387363066 +0200
@@ -11,4 +11,4 @@
 
 LIC_FILES_CHKSUM = "file://main.c;endline=27;md5=a2a70db58191379e2550cbed95449fbd"
 
-COMPATIBLE_MACHINE = '(sama5d3xek|sama5d3-xplained|sama5d3-xplained-sd|at91sam9x5ek|at91sam9rlek|at91sam9m10g45ek|sama5d4ek|sama5d4-xplained|sama5d4-xplained-sd|sama5d2-xplained|sama5d2-xplained-sd|sama5d2-xplained-emmc|sama5d2-ptc-ek|sama5d2-ptc-ek-sd|sama5d27-som1-ek|sama5d27-som1-ek-sd|sama5d2-icp-sd)'
+COMPATIBLE_MACHINE = '(sama5d3xek|sama5d3-xplained|sama5d3-xplained-sd|at91sam9x5ek|at91sam9rlek|at91sam9m10g45ek|sama5d4ek|sama5d4-xplained|sama5d4-xplained-sd|sama5d2-xplained|sama5d2-xplained-sd|sama5d2-xplained-emmc|sama5d2-ptc-ek|sama5d2-ptc-ek-sd|sama5d27-som1-ek|sama5d27-som1-ek-sd|sama5d2-icp-sd|sama5d27-hd96)'
diff -Naurw ./recipes-bsp/at91bootstrap/at91bootstrap.inc ../meta-atmel/recipes-bsp/at91bootstrap/at91bootstrap.inc
--- ./recipes-bsp/at91bootstrap/at91bootstrap.inc	2019-09-17 14:48:46.011177845 +0200
+++ ../meta-atmel/recipes-bsp/at91bootstrap/at91bootstrap.inc	2019-09-17 14:53:00.469989468 +0200
@@ -31,6 +31,7 @@
 AT91BOOTSTRAP_CONFIG_sama5d2-ptc-ek-sd ??= "${AT91BOOTSTRAP_MACHINE}sd_uboot"
 AT91BOOTSTRAP_CONFIG_sama5d27-som1-ek ??= "${AT91BOOTSTRAP_MACHINE}qspi_uboot"
 AT91BOOTSTRAP_CONFIG_sama5d27-som1-ek-sd ??= "${AT91BOOTSTRAP_MACHINE}sd_uboot"
+AT91BOOTSTRAP_CONFIG_sama5d27-hd96 ??= "${AT91BOOTSTRAP_MACHINE}sd_uboot"
 AT91BOOTSTRAP_CONFIG_sama5d2-icp-sd ??= "${AT91BOOTSTRAP_MACHINE}sd_uboot"
 
 AT91BOOTSTRAP_TARGET ??= "${AT91BOOTSTRAP_CONFIG}_defconfig"
@@ -41,6 +42,7 @@
 AT91BOOTSTRAP_LOAD_sama5d27-ptc-ek-sd ??= "sdboot-uboot"
 AT91BOOTSTRAP_LOAD_sama5d27-som1-ek ??= "qspiboot-uboot"
 AT91BOOTSTRAP_LOAD_sama5d27-som1-ek-sd ??= "sdboot-uboot"
+AT91BOOTSTRAP_LOAD_sama5d27-hd96 ??= "sdboot-uboot"
 AT91BOOTSTRAP_LOAD_sama5d3-xplained-sd ??= "sdboot-uboot"
 AT91BOOTSTRAP_LOAD_sama5d4-xplained-sd ??= "sdboot-uboot"
 AT91BOOTSTRAP_LOAD_sama5d2-icp-sd ??= "sdboot-uboot"
diff -Naurw ./recipes-bsp/dt-overlay-at91/dt-overlay-at91_git.bb ../meta-atmel/recipes-bsp/dt-overlay-at91/dt-overlay-at91_git.bb
--- ./recipes-bsp/dt-overlay-at91/dt-overlay-at91_git.bb	2019-09-17 14:48:46.011177845 +0200
+++ ../meta-atmel/recipes-bsp/dt-overlay-at91/dt-overlay-at91_git.bb	2019-09-17 14:50:58.387363066 +0200
@@ -6,7 +6,7 @@
 
 inherit deploy
 
-COMPATIBLE_MACHINE = '(sama5d3-xplained|sama5d3-xplained-sd|at91sam9x5ek|at91sam9x5ek-sd|sama5d4-xplained|sama5d4-xplained-sd|sama5d2-xplained|sama5d2-xplained-sd|sama5d2-xplained-emmc|sama5d2-ptc-ek|sama5d2-ptc-ek-sd|sama5d27-som1-ek|sama5d27-som1-ek-sd|sama5d2-icp-sd)'
+COMPATIBLE_MACHINE = '(sama5d3-xplained|sama5d3-xplained-sd|at91sam9x5ek|at91sam9x5ek-sd|sama5d4-xplained|sama5d4-xplained-sd|sama5d2-xplained|sama5d2-xplained-sd|sama5d2-xplained-emmc|sama5d2-ptc-ek|sama5d2-ptc-ek-sd|sama5d27-som1-ek|sama5d27-som1-ek-sd|sama5d2-icp-sd|sama5d27-hd96)'
 
 SRC_URI = "git://github.com/linux4sam/dt-overlay-at91.git;protocol=https"
 
diff -Naurw ./recipes-bsp/u-boot/files/envs/sama5d27-hd96.txt ../meta-atmel/recipes-bsp/u-boot/files/envs/sama5d27-hd96.txt
--- ./recipes-bsp/u-boot/files/envs/sama5d27-hd96.txt	1970-01-01 01:00:00.000000000 +0100
+++ ../meta-atmel/recipes-bsp/u-boot/files/envs/sama5d27-hd96.txt	2019-09-17 14:50:58.387363066 +0200
@@ -0,0 +1,7 @@
+bootargs=console=ttyS0,115200 root=/dev/mmcblk1p2 rw rootfstype=ext4 rootwait atmel.pm_modes=standby,ulp1 rng_core.default_quality=700
+bootcmd=fatload mmc 1:1 0x24000000 sama5d27_hd96.itb; bootm 0x24000000
+bootdelay=1
+ethact=gmac0
+stderr=serial
+stdin=serial
+stdout=serial
diff -Naurw ./recipes-bsp/u-boot/u-boot-at91_2019.04.bb ../meta-atmel/recipes-bsp/u-boot/u-boot-at91_2019.04.bb
--- ./recipes-bsp/u-boot/u-boot-at91_2019.04.bb	2019-09-17 14:48:46.011177845 +0200
+++ ../meta-atmel/recipes-bsp/u-boot/u-boot-at91_2019.04.bb	2019-09-17 14:50:58.387363066 +0200
@@ -10,7 +10,7 @@
 
 DEPENDS += "bison-native flex-native"
 
-COMPATIBLE_MACHINE = '(sama5d3xek|sama5d3-xplained|sama5d3-xplained-sd|at91sam9x5ek|at91sam9rlek|at91sam9m10g45ek|sama5d4ek|sama5d4-xplained|sama5d4-xplained-sd|sama5d2-xplained|sama5d2-xplained-sd|sama5d2-xplained-emmc|sama5d2-ptc-ek|sama5d2-ptc-ek-sd|sama5d27-som1-ek|sama5d27-som1-ek-sd|sama5d2-icp-sd)'
+COMPATIBLE_MACHINE = '(sama5d3xek|sama5d3-xplained|sama5d3-xplained-sd|at91sam9x5ek|at91sam9rlek|at91sam9m10g45ek|sama5d4ek|sama5d4-xplained|sama5d4-xplained-sd|sama5d2-xplained|sama5d2-xplained-sd|sama5d2-xplained-emmc|sama5d2-ptc-ek|sama5d2-ptc-ek-sd|sama5d27-som1-ek|sama5d27-som1-ek-sd|sama5d2-icp-sd|sama5d27-hd96)'
 
 UBRANCH = "u-boot-2019.04-at91"
 SRC_URI = "git://github.com/linux4sam/u-boot-at91.git;protocol=https;branch=${UBRANCH}"
diff -Naurw ./recipes-kernel/linux/linux-at91_4.19.bb ../meta-atmel/recipes-kernel/linux/linux-at91_4.19.bb
--- ./recipes-kernel/linux/linux-at91_4.19.bb	2019-09-17 14:48:46.011177845 +0200
+++ ../meta-atmel/recipes-kernel/linux/linux-at91_4.19.bb	2019-09-17 14:50:58.387363066 +0200
@@ -48,4 +48,4 @@
 
 KERNEL_MODULE_AUTOLOAD += "atmel_usba_udc g_serial"
 
-COMPATIBLE_MACHINE = "(sama5d2-xplained|sama5d2-xplained-sd|sama5d2-xplained-emmc|sama5d2-ptc-ek|sama5d2-ptc-ek-sd|sama5d27-som1-ek|sama5d27-som1-ek-sd|sama5d4-xplained|sama5d4-xplained-sd|sama5d4ek|sama5d3-xplained|sama5d3-xplained-sd|sama5d3xek|at91sam9x5ek|at91sam9m10g45ek|at91sam9rlek|sama5d2-icp-sd)"
+COMPATIBLE_MACHINE = "(sama5d2-xplained|sama5d2-xplained-sd|sama5d2-xplained-emmc|sama5d2-ptc-ek|sama5d2-ptc-ek-sd|sama5d27-som1-ek|sama5d27-som1-ek-sd|sama5d4-xplained|sama5d4-xplained-sd|sama5d4ek|sama5d3-xplained|sama5d3-xplained-sd|sama5d3xek|at91sam9x5ek|at91sam9m10g45ek|at91sam9rlek|sama5d2-icp-sd|sama5d27-hd96)"
