diff -Naurw ./board/sama5d27_som1_ek/sama5d27_som1_eksd_uboot_defconfig ../git/board/sama5d27_som1_ek/sama5d27_som1_eksd_uboot_defconfig
--- ./board/sama5d27_som1_ek/sama5d27_som1_eksd_uboot_defconfig	2019-02-26 16:04:06.207300562 +0100
+++ ../git/board/sama5d27_som1_ek/sama5d27_som1_eksd_uboot_defconfig	2019-02-26 16:14:50.303300562 +0100
@@ -1,6 +1,7 @@
 CONFIG_SAMA5D27_SOM1_EK=y
 CONFIG_RAM_128MB=y
 CONFIG_SDCARD=y
+CONFIG_SDHC1=y
 CONFIG_JUMP_ADDR="0x23f00000"
 CONFIG_DEBUG=y
 # CONFIG_ENTER_NWD is not set
diff -Naurw ./driver/sdhc.c ../git/driver/sdhc.c
--- ./driver/sdhc.c	2018-11-12 17:23:36.000000000 +0100
+++ ../git/driver/sdhc.c	2019-02-21 18:29:56.491278269 +0100
@@ -493,6 +493,8 @@
 	return 0;
 }
 
+#if 0
+
 static int sdhc_is_card_inserted(struct sd_card *sdcard)
 {
 	/*
@@ -539,6 +541,13 @@
 	return is_inserted;
 }
 
+#else
+
+#define sdhc_is_card_inserted( x )  1
+
+#endif
+
+
 static int sdhc_init(struct sd_card *sdcard)
 {
 	unsigned int normal_status_mask, error_status_mask;
@@ -747,6 +756,8 @@
 		}
 	}
 
+	sdhc_writeb( SDMMC_MC1R, sdhc_readb(SDMMC_MC1R) | 0x80 );
+
 	sdhc_writel(SDMMC_ARG1R, sd_cmd->argu);
 
 	sdhc_writew(SDMMC_CR, cmd_reg);
