diff -Naurw ./arch/arm/boot/dts/at91-sama5d27_som1.dtsi ../kernel-source/arch/arm/boot/dts/at91-sama5d27_som1.dtsi
--- ./arch/arm/boot/dts/at91-sama5d27_som1.dtsi	2018-11-12 15:56:40.000000000 +0100
+++ ../kernel-source/arch/arm/boot/dts/at91-sama5d27_som1.dtsi	2019-02-27 14:17:14.115544616 +0100
@@ -56,7 +56,7 @@
 		};
 
 		main_xtal {
-			clock-frequency = <24000000>;
+			clock-frequency = <12000000>;
 		};
 	};
 
diff -Naurw ./arch/arm/boot/dts/at91-sama5d27_som1_ek.dts ../kernel-source/arch/arm/boot/dts/at91-sama5d27_som1_ek.dts
--- ./arch/arm/boot/dts/at91-sama5d27_som1_ek.dts	2018-11-12 15:56:40.000000000 +0100
+++ ../kernel-source/arch/arm/boot/dts/at91-sama5d27_som1_ek.dts	2019-02-27 14:18:04.775544616 +0100
@@ -99,6 +99,7 @@
 			bus-width = <4>;
 			pinctrl-names = "default";
 			pinctrl-0 = <&pinctrl_sdmmc1_default>;
+			cd-gpios = <&pioA PIN_PD23 GPIO_ACTIVE_HIGH>;
 			status = "okay";
 		};
 
diff -Naurw ./drivers/mmc/host/sdhci-of-at91.c ../kernel-source/drivers/mmc/host/sdhci-of-at91.c
--- ./drivers/mmc/host/sdhci-of-at91.c	2018-11-12 15:56:45.000000000 +0100
+++ ../kernel-source/drivers/mmc/host/sdhci-of-at91.c	2019-02-21 18:14:42.671278269 +0100
@@ -125,7 +125,7 @@
 {
 	sdhci_reset(host, mask);
 
-	if (host->mmc->caps & MMC_CAP_NONREMOVABLE)
+	/* if (host->mmc->caps & MMC_CAP_NONREMOVABLE) */
 		sdhci_at91_set_force_card_detect(host);
 }
 
@@ -412,7 +412,7 @@
 	 * This bit is reset when a software reset for all command is performed
 	 * so we need to implement our own reset function to set back this bit.
 	 */
-	if (host->mmc->caps & MMC_CAP_NONREMOVABLE)
+	/* if (host->mmc->caps & MMC_CAP_NONREMOVABLE) */
 		sdhci_at91_set_force_card_detect(host);
 
 	device_init_wakeup(&pdev->dev, true);
